agent/php_execute.c

```diff
@@ -1036,18 +1051,31 @@ static void nr_php_execute_metadata_add_code_level_metrics(
     /*
      * We are getting CLM for a function.
      */
-    lineno = nr_php_zend_execute_data_lineno(execute_data);
+    if (0 == metadata->function_lineno) {
+      lineno = nr_php_zend_execute_data_lineno(execute_data);
+    } else {
+      /*
+       * If the metadata was already set (in the case of OAPI where we need to
+       * get it preemptively, use that value for lineno instead.
+       */
+      lineno = metadata->function_lineno;
+    }
   }
-  if (nr_strempty(function)) {
+
+#define CHK_CLM_EMPTY(s) ((NULL == s || nr_strempty(s)) ? true : false)
```

Comment:
Seems like a good utility function to have if nr_strempty() isn't reliable for this test.

Reply:
`nr_strempty` is NULL-safe - the first thing it checks is if its argument is NULL. I'd also like to understand why this extra `NULL` check is done before calling NULL-safe function.